src/langrila/memory/s3.py

```diff
@@ -0,0 +1,84 @@
+import os
+import logging
+import json
+from typing import Optional, Union
+
+import boto3
+from botocore.exceptions import ClientError
+from botocore.client import Config as BotoConfig
+
+from ..base import BaseConversationMemory
+
+class S3ConversationMemory(BaseConversationMemory):
+    def __init__(
+            self,
+            bucket: str,
+            key: str,
+            *,
+            region_name: Optional[str] = None,
+            api_version: Optional[str] = None,
+            use_ssl: Optional[bool] = True,
+            verify: Union[str, bool, None] = None,
+            endpoint_url: Optional[str] = None,
+            aws_access_key_id: Optional[str] = None,
+            aws_secret_access_key: Optional[str] = None,
+            aws_session_token: Optional[str] = None,
```

Comment:
Initially I thought some of these arguments can be specified by environmental variable in the same manner as other modules, but could other issue happen?

Reply:
Using temporary credentials such as IAM roles is a best practice instead of access key and secret key. For me, using these keys here is like a option for simple testing purpose and thought it should be better to leave these to users.

but as you say, unifying in the same manners among modules is better and easier to understand. will fix.